vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ColumnResizeEvent.java

```diff
@@ -0,0 +1,93 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.grid;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.vaadin.flow.component.ComponentEvent;
+import com.vaadin.flow.component.DomEvent;
+import com.vaadin.flow.component.EventData;
+import com.vaadin.flow.component.grid.Grid.Column;
+
+import elemental.json.JsonObject;
+
+/**
+ * Event fired when a Grid column is resized by the user.
+ *
+ * @param <T>
+ *            the grid bean type
+ *
+ * @author Vaadin Ltd
+ *
+ * @see Grid#addColumnResizeListener(com.vaadin.flow.component.ComponentEventListener)
+ *
+ */
+@DomEvent("column-drag-resize")
+public class ColumnResizeEvent<T> extends ComponentEvent<Grid<T>> {
+
+    private final JsonObject columnWidths;
+    private final JsonObject columnFlexGrows;
+    private final String resizedColumnKey;
+
+    /**
+     * Creates a new column resize event.
+     *
+     * @param source
+     *            the component that fired the event
+     * @param fromClient
+     *            <code>true</code> if the event was originally fired on the
+     *            client, <code>false</code> if the event originates from
+     *            server-side logic
+     * @param columnWidths
+     *            CSS width properties of all visible columns when resizing
+     *            finished, mapped by their internal id
+     * @param columnFlexGrows
+     *            CSS flex-grow properties of all visible columns when resizing
+     *            finished, mapped by their internal id
+     * @param resizedColumnKey
+     *            internal id of the column that was the target of user's resize
+     *            action
+     *
+     */
+    public ColumnResizeEvent(Grid<T> source, boolean fromClient,
+            @EventData("event.detail.columnWidths") JsonObject columnWidths,
+            @EventData("event.detail.columnFlexGrows") JsonObject columnFlexGrows,
+            @EventData("event.detail.resizedColumnKey") String resizedColumnKey) {
+        super(source, fromClient);
+        this.columnWidths = columnWidths;
+        this.columnFlexGrows = columnFlexGrows;
+        this.resizedColumnKey = resizedColumnKey;
+    }
+
+    public Column<T> getResizedColumn() {
+        return getSource().getColumns().stream()
+                .filter(col -> col.getInternalId().equals(resizedColumnKey))
+                .findFirst().orElse(null);
+    }
+
+    public Map<Column<T>, String> getColumnWidths() {
```

Comment:
Missing javadocs

Reply:
Added javadocs

Reply:
Changed test to check that correct flex grows and three column widths are returned in the event